src/domain_bridge/domain_bridge.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright 2021, Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "domain_bridge/domain_bridge.hpp"
+
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "domain_bridge/exceptions.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/executors/multi_threaded_executor.hpp"
+#include "rcutils/logging_macros.h"
+#include "rosbag2_cpp/typesupport_helpers.hpp"
+
+#include "generic_publisher.hpp"
+#include "generic_subscription.hpp"
+
+namespace domain_bridge
+{
+
+/// Implementation of \ref DomainBridge.
+class DomainBridgeImpl
+{
+public:
+  using SubscriptionMap =
+    std::map<std::pair<size_t, std::string>, std::shared_ptr<GenericSubscription>>;
+  using PublisherMap = std::map<std::pair<size_t, std::string>, std::shared_ptr<GenericPublisher>>;
+  using NodeMap = std::map<size_t, std::shared_ptr<rclcpp::Node>>;
+
+  DomainBridgeImpl()
+  {
+    this->executor_context_ = create_context_with_domain_id(1);
+    rclcpp::ExecutorOptions executor_options;
+    executor_options.context = this->executor_context_;
+    this->executor_ = std::make_unique<rclcpp::executors::MultiThreadedExecutor>(executor_options);
+  }
+
+  ~DomainBridgeImpl()
+  {
+    const std::string reason("Domain bridge shutdown");
+    this->executor_context_->shutdown(reason);
```

Comment:
```suggestion
    this->executor_context_->shutdown("Domain bridge shutdown");
```

?

Reply:
This was vestigial, and removed in the latest changes anyways.